src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java

```diff
@@ -1255,6 +1285,19 @@ protected void wipeOfTestResources(
         }
     }
 
+    protected void validateDocCountAndEmbedding(String indexName, int expectedDocCount, String docId, final String embeddingField) {
+        int count = getDocCount(indexName);
+        assertEquals(expectedDocCount, count);
+        Map<String, Object> document = getDocById(indexName, docId);
+        Object documentSource = document.get("_source");
+        assertTrue(documentSource instanceof Map);
+        @SuppressWarnings("unchecked")
+        Map<String, Object> documentSourceMap = (Map<String, Object>) documentSource;
+        assertTrue(documentSourceMap.containsKey(embeddingField));
+        Object ingestOutputs = documentSourceMap.get(embeddingField);
+        assertTrue(ingestOutputs instanceof Map);
+    }
```

Comment:
Thanks for creating this function in the base class. How about making this function more general? The function will only check whether a certain field exists for every document in the index. If so, it will return a list with elements in this field. The returned list can be used for further evaluation.

Reply:
That's a good callout. But I feel it makes a bit more sense that we modify it to the recommended manner when we do have the need, it won't be a fully restructure but a couple of lines change.

Reply:
In fact, this function can be used by text chunking BWC tests files. Can you take a look at the `validateTestIndex` function? These two functions are pretty similar and we can reuse them.

Reply:
Refactored some code to reuse the main function

Reply:
Great! Thank you.